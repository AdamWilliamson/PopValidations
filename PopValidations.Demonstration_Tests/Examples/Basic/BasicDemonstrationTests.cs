using ApprovalTests;
using PopValidations_Tests.TestHelpers;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations.Demonstration_Tests.Examples.Basic;

public class BasicDemonstrationTests
{
    [Fact]
    public async Task Validation()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());
        var song = new BasicSong(
            "Disturbed",
            null,
            "Down With The Sickness",
            2.4,
            string.Empty
        );

        // Act
        var results = await runner.Validate(song);

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(results));
    }

    [Fact]
    public void Description()
    {
        // Arrange
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new BasicSongValidator());

        // Act
        var results = runner.Describe();

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(results));
    }
}

