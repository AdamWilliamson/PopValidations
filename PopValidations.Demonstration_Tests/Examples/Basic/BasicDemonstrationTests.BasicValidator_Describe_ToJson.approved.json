{
    "results": [
        {
            "property": "TrackNumber",
            "outcomes": [
                {
                    "validator": "NotNullValidation",
                    "message": "Must not be null.",
                    "values": []
                }
            ],
            "validationGroups": []
        },
        {
            "property": "TrackName",
            "outcomes": [
                {
                    "validator": "IsEqualToValidation",
                    "message": "Must equal to 'Definitely Not The Correct Song Name.'",
                    "values": [
                        {
                            "key": "value",
                            "value": "Definitely Not The Correct Song Name."
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "Duration",
            "outcomes": [
                {
                    "validator": "IsEqualToValidation",
                    "message": "Songs must force you to travel slowly backwards in time to listen to.",
                    "values": [
                        {
                            "key": "value",
                            "value": "-1"
                        }
                    ]
                }
            ],
            "validationGroups": []
        },
        {
            "property": "Genre",
            "outcomes": [
                {
                    "validator": "IsNotEmptyValidation",
                    "message": "Must not be empty.",
                    "values": []
                },
                {
                    "validator": "IsLengthInclusivelyBetweenValidation",
                    "message": "Must be between 20 and 400 inclusive.",
                    "values": [
                        {
                            "key": "startValue",
                            "value": "20"
                        },
                        {
                            "key": "endValue",
                            "value": "400"
                        }
                    ]
                }
            ],
            "validationGroups": []
        }
    ]
}